### PR TITLE
[reciever/filelog] - Adding missing Register

### PR DESCRIPTION
**Pull request by rehanpfmr**
**Description:** <Describe what has changed.>
Adding the key_value_parser operator- This was never registered, so cannot be used as an operator in the filelog parser receiver.
The key_value_parser operator parses the string-type field selected by parse_from into key value pairs. 

**Link to tracking Issue:** (https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/13631)

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** Added document URL in pkg/stanza/docs/operators/README.md

### CODE REVIEW

**Comment by djaglowski** [comment]
Looks good. Please add a changelog entry.

**Review by rehanpfmr — COMMENTED** [review]
@djaglowski I have added the Changelog file.

File: CHANGELOG.md [hunk 1]
```diff
@@ -46,6 +46,7 @@ in outgoing logs data.
 - `jaegerreceiver`: Add remote sampling deprecation warning (#6633)
 
 ### 🚀 New components 🚀
+- `keyvalueparser`: Add key_value_parser (#13631)
```
[the remark this answers is not in the document]
Comment (rehanpfmr) [reply CHANGELOG.md]: Ahh!, I have removed the changes from CHANGELOG. I hope this should now resolve.

**Comment by djaglowski** [comment]
Please address lint failure

**Review by rehanpfmr — COMMENTED** [review]
Tried to fix it by running "gofmt -s -w pkg/stanza/adapter/register.go"

**Comment by rehanpfmr** [comment]
@djaglowski would you please review?

**Review by djaglowski — APPROVED** [review]
LGTM. I've approved the workflow, so this can be merged if checks pass.

**Comment by rehanpfmr** [comment]
Thanks @djaglowski!

**Pull request merged**